lib/functions.php

```diff
@@ -62,7 +62,10 @@ function resolve(string $basePath, string $newPath): string
             $path .= '/'.$delta['path'];
         }
     } else {
-        $path = $base['path'] ?: '/';
+        $path = $base['path'] ?? '/';
+        if ('' === $path) {
+            $path = '/';
+        }
```

Comment:
phpstan strict rules do not like use of the "?:" operator. It suggests using the "??" null coalesce operator or the full ternary operator. Actually that is "a good thing", because in this case if `$base['path']` is the string "0" then `?:` sees that as false and takes the "/", which results in the "0" path disappearing from the URI - that is an existing (very edge case) bug.

I added unit test cases to cover various permutations of the base path being "0" "a" or an empty string. One of the "0" cases fails prior to this code change, and now passes.

The interesting different cases for `$base['path']` are when it is not set (`null`) - that is covered by the `??` null coalesce operator - "0" which now works, and the empty string - which now has a specific check.

The wonders of PHP!